components/dialog/RotateEncryptionKeyDialog.vue

```diff
@@ -64,8 +72,19 @@ export default {
     },
 
     async apply(buttonDone) {
+      const isRke2 = this.cluster.isRke2;
+
       try {
-        await this.cluster.mgmt.doAction('rotateEncryptionKey');
+        if (isRke2) {
+          const currentGeneration = this.cluster.spec?.rkeConfig?.rotateCertificates?.generation || 0;
+
+          // To rotate the encryption keys, increment
+          // rkeConfig.rotateEncyrptionKeys.generation in the YAML.
+          set(this.cluster, 'spec.rkeConfig', { rotateEncryptionKeys: { generation: currentGeneration + 1 } });
```

Comment:
We may want to switch to the existing [`set()` Lodash helper](https://lodash.com/docs/4.17.15#set)?

```js
set(this.cluster, 'spec.rkeConfig.rotateCertificates.generation', currentGeneration + 1)
```

Reply:
I have replaced the lodash helper because the lodash one creates the path if it doesn't already exist, making it look cleaner.